### PR TITLE
Initial bringup to 1.21.0

### PR DESCRIPTION
**Pull request by JonnyVR1**
There are still some things that don't work - I don't fully understand C++ as I'm a C# dev.  The remaining issues should be fairly quick to solve for someone who knows what they're doing though!

My main issue atm is that level->get_levelID() seems to cause a crash and I have no idea why so I can't really progress any further.

### CODE REVIEW

**Comment by JonnyVR1** [comment]
Fixed the level->get_levelID() crash, now the issue is that when filtering (e.g. FilterOriginal method) we get cannot find method errors in ArrayUtil.cpp, despite the methods/fields being present in codegen List implementation.

File: qpm_defines.cmake [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# YOU SHOULD NOT MANUALLY EDIT THIS FILE, QPM WILL VOID ALL CHANGES
```
Comment (RedBrumbler) [inline qpm_defines.cmake:1]: This file should not be uploaded on the repo

File: include/static-defines.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (RedBrumbler) [inline include/static-defines.h:1]: you're doing something wrong if you need to define these, I can't merge this

Comment (RedBrumbler) [reply include/static-defines.h:1]: File should probably be removed altogether

File: extern.cmake [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# YOU SHOULD NOT MANUALLY EDIT THIS FILE, QPM WILL VOID ALL CHANGES
```
Comment (RedBrumbler) [inline extern.cmake:1]: This file also should not be on the repo

File: Android.mk [hunk 4]
```diff
@@ -21,61 +21,61 @@ rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
 # Creating prebuilt for dependency: modloader - version: 1.2.3
```
Comment (RedBrumbler) [inline Android.mk:21]: Android mk and application mk are not needed anymore for qpm rust, these should be removed from the repo

(git rm ./Android.mk && git rm ./Application.mk)

File: qpm.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "info": {
     "name": "SongBrowser",
     "id": "songbrowser",
-    "version": "0.1.0",
+    "version": "1.0.2",
```
Comment (RedBrumbler) [inline qpm.json:7]: We don't actually update our versions directly, we let the GH action on release do that, this is not needed

File: src/UI/Browser/SongButtons.cpp [hunk 6]
```diff
@@ -1,4 +1,4 @@
 #include "UI/Browser/SongButtons.hpp"
 
-DEFINE_TYPE(SongBrowser::UI, SongFilterButton);
-DEFINE_TYPE(SongBrowser::UI, SongSortButton);
\ No newline at end of file
+DEFINE_TYPE(SongBrowser::UI, SongFilterButton)
```
Comment (RedBrumbler) [inline src/UI/Browser/SongButtons.cpp:3]: even if the semicolons are not needed I'd prefer to keep them

File: src/logging.cpp [hunk 7]
```diff
@@ -1,6 +1,7 @@
 #include "logging.hpp"
 #include <map>
 #include <string>
+#include "static-defines.h"
```
Comment (RedBrumbler) [inline src/logging.cpp:4]: ties into "static defines file not needed" but this should not be here

File: src/main.cpp [hunk 8]
```diff
@@ -25,6 +23,7 @@
 #include "GlobalNamespace/MenuTransitionsHelper.hpp"
 #include "Zenject/DiContainer.hpp"
 #include "System/Action_1.hpp"
+#include "static-defines.h"
```
Comment (RedBrumbler) [inline src/main.cpp:26]: ties into "static defines file not needed" but this should not be here

**Review by RedBrumbler — CHANGES_REQUESTED** [review]
I can't merge the PR like this as it'd clutter the repo too much, I think what mainly should be done is keep to just the src and include edits, and even then mostly the src ones

File: include/static-defines.h (hunk 2 above)
Comment (JonnyVR1) [reply include/static-defines.h:1]: Disagree, mod won't compile without this because of references to ID

File: src/logging.cpp (hunk 7 above)
Comment (JonnyVR1) [reply src/logging.cpp:4]: As above, mod won't compile without this because of references to ID

File: src/main.cpp (hunk 8 above)
Comment (JonnyVR1) [reply src/main.cpp:26]: As above, mod won't compile without this because of references to ID

File: include/static-defines.h (hunk 2 above)
Comment (RedBrumbler) [reply include/static-defines.h:1]: then something is going wrong, as these defines really are not needed if compilation is setup right

**Comment by RedBrumbler** [comment]
I'm missing the CMakeLists.txt file, I think that ties into why ID is needed instead of what is now used (MOD_ID)

File: include/static-defines.h (hunk 2 above)
Comment (JonnyVR1) [reply include/static-defines.h:1]: By all means, try to compile it without that include. If you can get it to work then I'd like to know the steps as for me I had to add this line to get it to compile.

**Comment by JonnyVR1** [comment]
> I'm missing the CMakeLists.txt file, I think that ties into why ID is needed instead of what is now used (MOD_ID)

In that case then this isn't in the remit of this PR to fix.

**Comment by JonnyVR1** [comment]
Switched to MOD_ID

**Comment by Christoffyw** [comment]
The powershell scripts are outdated. They are still using Android.mk

**Comment by RedBrumbler** [comment]
ahhh that must be why it wasn't properly building n stuff

**Comment by JonnyVR1** [comment]
> The powershell scripts are outdated. They are still using Android.mk

Outside of the remit of this PR

**Comment by JonnyVR1** [comment]
For funsies I've updated the build setup to the new standard, even though it was outside of the remit of this PR. cba arguing about this in the comments so just easier to update it.

**Comment by RedBrumbler** [comment]
ahhh yea I guess in technicality it's outside the PR but all the tooling uses that now so it would be harder not to do that

**Review by RedBrumbler — APPROVED** [review]

**Pull request merged**